src/main/modules/ardoq/ArdoqClient.ts

```diff
@@ -1,73 +1,185 @@
 import { Dependency } from './Dependency';
 
 import { AxiosInstance } from 'axios';
+import config from 'config';
+
+const { Logger } = require('@hmcts/nodejs-logging');
 
 export class ArdoqClient {
+  readonly componentTypeLookup = new Map<ArdoqWorkspace, string>([
+    [ArdoqWorkspace.ARDOQ_VCS_HOSTING_WORKSPACE, 'p1681283498700'],
+    [ArdoqWorkspace.ARDOQ_CODE_REPOSITORY_WORKSPACE, 'p1681283498700'],
+    [ArdoqWorkspace.ARDOQ_SOFTWARE_FRAMEWORKS_WORKSPACE, 'p1659003743296'],
+  ]);
+
   constructor(
     private httpClient: AxiosInstance,
-    private apiWorkspace: string,
-    private cache: Map<string, string> = new Map<string, string>()
+    private cache: Map<string, Dependency> = new Map<string, Dependency>(),
+    private logger = Logger.getLogger('ArdoqClient')
   ) {}
 
   private cacheResult(d: Dependency) {
-    this.cache.set(d.name, d.version);
+    this.cache.set(d.name, d);
   }
 
   private isCached(d: Dependency): boolean {
-    return this.cache.get(d.name) === d.version;
+    const found = this.cache.get(d.name);
+    if (found === undefined) {
+      return false;
+    }
+    return found.equals(d);
   }
 
-  private searchForComponent(dependencyFullName: string) {
-    return this.httpClient.get('/api/component/search', {
+  private searchForComponent(componentName: string, workspace: ArdoqWorkspace) {
+    return this.httpClient.get('/api/v2/components', {
       params: {
-        workspace: this.apiWorkspace,
-        name: dependencyFullName,
+        rootWorkspace: config.get(workspace),
+        name: componentName,
       },
       responseType: 'json',
     });
   }
 
-  private createComponent(dependencyFullName: string) {
+  private createComponent(componentName: string, workspace: ArdoqWorkspace) {
     return this.httpClient.post(
-      '/api/component',
+      '/api/v2/components',
       {
-        rootWorkspace: this.apiWorkspace,
-        name: dependencyFullName,
+        rootWorkspace: config.get(workspace),
+        name: componentName,
+        typeId: this.componentTypeLookup.get(workspace),
       },
       {
         params: {
-          workspace: this.apiWorkspace,
-          name: dependencyFullName,
+          rootWorkspace: config.get(workspace),
+          name: componentName,
+          typeId: this.componentTypeLookup.get(workspace),
         },
         responseType: 'json',
       }
     );
   }
 
-  async updateDep(d: Dependency): Promise<ArdoqComponentCreatedResponse> {
-    if (this.isCached(d)) {
-      return ArdoqComponentCreatedResponse.EXISTING;
-    }
+  private createOrGetComponent(name: string, workspace: ArdoqWorkspace): Promise<string | null> {
+    return this.searchForComponent(name, workspace).then(searchRes => {
+      if (searchRes.status === 200 && searchRes.data.values.length > 0) {
+        this.logger.debug('Found component: ' + name);
+        return searchRes.data.values[0]._id;
+      }
+      return this.createComponent(name, workspace).then(createRes => {
```

Comment:
I would make the method async and then await:

```
const createRes = await this.createComponent(name, workspace);
...
```

Control flow is easier to follow without the inline function